pkg/kubelet/cm/container_manager_stub.go

```diff
@@ -71,10 +76,38 @@ func (cm *containerManagerStub) NewPodContainerManager() PodContainerManager {
 	return &podContainerManagerStub{}
 }
 
-func (cm *containerManagerStub) GetResources(pod *v1.Pod, container *v1.Container, activePods []*v1.Pod) (*kubecontainer.RunContainerOptions, error) {
+func (cm *containerManagerStub) GetResources(pod *v1.Pod, container *v1.Container) (*kubecontainer.RunContainerOptions, error) {
 	return &kubecontainer.RunContainerOptions{}, nil
 }
 
+// UpdatePluginResources makes a simple comparison between resources required in pod
```

Comment:
I think container_manager_stub is used at a lot of places and is supposed to be no-op for most of its operations. The behavior defined here seems specific to a single test. Can we instead define a function, say updatePluginResourcesFunc() inside TestHandlePluginResources()? This way, it is also easier to define the function behavior directly related to the test. E.g., instead of using resourceName1 and resourceName2, we can use adjustedResource and unadjustedResouce and have updatePluginResourcesFunc() simply adjust nodeinfo allocatable values for adjustedResource.

Reply:
Other than testing resource adjustment, can we also add another resource, say failedResource, upon which updatePluginResourcesFunc() simply returns an error and verifies that pod requiring that resource would fail?

Also forgot to mention above that after defining updatePluginResourcesFunc() inside TestHandlePluginResources(), we can add it with:
kl.admitHandlers.AddPodAdmitHandler(lifecycle.NewPredicateAdmitHandler(kubelet.getNodeAnyWay, lifecycle.NewAdmissionFailureHandlerStub(), updatePluginResourcesFunc))

Reply:
Well, I thought about these issues too when adding the test case, What I was concerned about is the place to add the PodAdmitHandler this way.
Do you suggest to add the handler in shared func **newTestKubelet** or especially in the new case TestHandlePluginResources()? The former may afftec other cases as well; the latter, however, may add lifecycle.NewPredicateAdmitHandler twice, which has no effect on the expected results though

Reply:
> Other than testing resource adjustment, can we also add another resource, say failedResource, upon which updatePluginResourcesFunc() simply returns an error and verifies that pod requiring that resource would fail?

I'm not quite sure about the behavior about **failedResource**  here. What's the difference between  **unadjustedResouce** and **failedResource**? You mean we need a resource that would fast fail, without comparison？ Do we have some corresponding practical scenarios?

Reply:
I mean the later, i.e., add it in the test case. It is ok to add the PredicateAdmitHandler twice. The one added by default is a no-op. In the test case, we will test the new behavior.

For the suggested failedResource, we can have updatePluginResourceFunc() to return error for the pod requiring this resource to emulate the allocation failure behavior.

Reply:
done :)